sig-security-external-audit/security-audit-2024/RFP.md

```diff
@@ -0,0 +1,796 @@
+   </td>
+   <td><strong>30 points</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><strong>Past Performance</strong>
+   </td>
+   <td>Have we worked with this team before? Were projects completed on schedule? Were the results in line with expectations?
+   </td>
+   <td><strong>10 points</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><strong>Documentation Quality</strong>
+   </td>
+   <td>All OSTIF work is done in public. Audit document quality is crucially important because the public is reviewing the overall quality of our published work continuously.
+   </td>
+   <td><strong>10 points</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td><strong>Total possible score**</strong>
+   </td>
+   <td><strong>100 points</strong>
+   </td>
+  </tr>
+</table>
+
+
+**In the unlikely event of a tie, diversity and inclusion of the audit team is considered for race, gender, and ethnic background to help promote a healthy and diverse professional environment for all.
+
+
+<table>
+  <tr>
+   <td><strong>Proposed Project: K8s</strong>
+   </td>
+   <td><strong>Maximum Budget: TBD</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>Github: Many - See below
+<p style="text-align: center">
+Website: <a href="https://kubernetes.io/">https://kubernetes.io/</a>
+   </td>
+   <td><em>Note: Budget is strongly considered as a criterion for project selection. The maximum budget is to give firms guidance on the relative expectations of the client.</em>
+   </td>
+  </tr>
+</table>
+
+
+
+<table>
+  <tr>
+   <td><strong>Language(s)</strong>
+<p style="text-align: center">
+Go
+   </td>
+   <td><strong>License(s)</strong>
+<p style="text-align: center">
+Apache 2.0
+   </td>
+  </tr>
+</table>
+
+
+**Overall Project Description**
+
+Kubernetes, also known as K8s, is an open source system for automating deployment, scaling, and management of containerized applications. Kubernetes is the premiere platform for open source containerized computing and a cornerstone of the global cloud computing industry. This security review focuses on specific components and subprojects that were out-of-scope in previous K8s audits, giving crucial insight into the security and health of these projects.
+
+Below are the components identified by the Kubernetes community as both needing additional security scrutiny and whose maintainers are open to enthusiastically participating in this project.
+
+These projects are ranked by importance on both the perceived criticality of the component by the community, and the information level of the documentation provided. It should also be carefully noted that auditors will have the flexibility to investigate hunches and seek out issues in other components if they so choose. The prioritization listed here is only compulsory for components specifically named below, or projects listed as “medium” and “high” priority in the [2024 Kubernetes Audit Scope document](https://docs.google.com/spreadsheets/d/1b_OrTHzLcPisShcera9eKtycKZORvW1z8Hm2If4NCF4).
```

Comment:
What exactly is meant by "The prioritization listed here is only compulsory for components specifically named below?" There is a list of 19 components that follows.

Is it expected that each proposal's scope be a subset of this list starting at the top and working its way down?

Are vendors free to choose a subset of these 19 components to scope their proposal, based on the expertise and time commitment available among their staff?

Reply:
I can reword this section to get the intent across. Basically we think that there is not enough time available to the researchers to get to all 90+ items, especially considering some of them are large and very complex. This wording is to prioritize the specific items named, but for all other subprojects/features it will be a best-effort approach to get as far as you can, and also investigate the components that the researchers find the most interesting. It also calls for the researchers to carefully call out what was covered by this review, and to specifically call out what they couldn't get to. This helps future independent researchers know where eyes have been and what hasn't been looked at in any follow-up work.

I'll reword this to be more clear.

Reply:
thank you! That'll be really helpful, I think.

The previous RFP may be useful as a source of inspiration and wording: https://github.com/kubernetes/sig-security/blob/main/sig-security-external-audit/security-audit-2021-2022/RFP.md

Reply:
Ack

I will review it while rewording that section.

Thank you!

Reply:
@tabbysable , I've updated this PR with updates from @OSTIF-Derek